src/main/java/com/treasuredata/android/TreasureData.java

```diff
@@ -873,6 +893,44 @@ public void enableAutoAppendLocaleInformation() {
         this.autoAppendLocaleInformation = true;
     }
 
+    public void enableAutoAppendAdvertisingIdentifier() {
+        enableAutoAppendAdvertisingIdentifier(EVENT_KEY_ADVERTISING_IDENTIFIER);
+    }
+
+    public void enableAutoAppendAdvertisingIdentifier(String columnName) {
```

Comment:
Just an idea, cc @kietdo360 
The class (`AdvertisingIdClient`) existent checking could be done eagerly. We could return a boolean whether advertising ID is enabled successfully or not here. Otherwise, applications have no way to check that have except look at the actual events, I assumed.

Reply:
I did consider your idea before. But realistically speaking, if a developer call this `enableAutoAppendAdvertisingIdentifier`, he must want to use this feature, hence he want to add Google Play Service Ads library. If he does not include that library, it should be handled as an error/Exception case, which is currently being handled inside `GetAdvertisingIdAsyncTask`.